tools/stonith_admin.c

```diff
@@ -323,103 +329,52 @@ handle_level(stonith_t *st, char *target, int fence_level,
                                        name, value, fence_level);
 }
 
-static char *
-fence_action_str(const char *action)
-{
-    char *str = NULL;
-
-    if (action == NULL) {
-        str = strdup("unknown");
-    } else if (action[0] == 'o') { // on, off
-        str = crm_concat("turn", action, ' ');
-    } else {
-        str = strdup(action);
-    }
-    return str;
-}
-
-static void
-print_fence_event(stonith_history_t *event)
-{
-    char *action_s = fence_action_str(event->action);
-    time_t complete = event->completed;
-
-    printf("%s was able to %s node %s on behalf of %s from %s at %s\n",
-           (event->delegate? event->delegate : "This node"), action_s,
-           event->target, event->client, event->origin, ctime(&complete));
-    free(action_s);
-}
-
 static int
-handle_history(stonith_t *st, const char *target, int timeout, int quiet,
-             int verbose, int cleanup, int broadcast)
+handle_history(stonith_t *st, const char *target, int timeout,
+             int verbose, int cleanup, int broadcast, pcmk__output_t *out)
 {
     stonith_history_t *history = NULL, *hp, *latest = NULL;
     int rc = 0;
 
-    if (!quiet) {
-        if (cleanup) {
-            printf("cleaning up fencing-history%s%s\n",
-                   target?" for node ":"", target?target:"");
-        }
-        if (broadcast) {
-            printf("gather fencing-history from all nodes\n");
-        }
+    if (cleanup) {
+        fprintf(stderr, "cleaning up fencing-history%s%s\n",
+                target?" for node ":"", target?target:"");
     }
+    if (broadcast) {
+        fprintf(stderr, "gather fencing-history from all nodes\n");
+    }
```

Comment:
Rather than print to stderr, I see a need for an output function for something that's useful for an interactive session (text formatter would print it) but an XML processor wouldn't care about (XML formatter would ignore it). Something like "progress", "hint", "informational", "what I'm going to do right now" ...